### PR TITLE
fix(server): change password with token should be public

### PR DESCRIPTION
**Pull request by forehalo**

### CODE REVIEW

**Comment by forehalo** [comment]
* **#7863** <a href="https://app.graphite.dev/github/pr/toeverything/AFFiNE/7863?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#7857** <a href="https://app.graphite.dev/github/pr/toeverything/AFFiNE/7857?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#7855** <a href="https://app.graphite.dev/github/pr/toeverything/AFFiNE/7855?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#7852** <a href="https://app.graphite.dev/github/pr/toeverything/AFFiNE/7852?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `canary`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @forehalo and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

**Review by JimmFly — APPROVED** [review]

File: packages/frontend/core/src/pages/auth.tsx [hunk 1]
```diff
@@ -98,22 +98,26 @@ export const AuthPage = (): ReactElement | null => {
...
+  if (!passwordLimits) {
     // TODO(@eyhn): loading UI
     return null;
   }
 
   switch (authType) {
     case 'onboarding':
-      return <OnboardingPage user={account} onOpenAffine={onOpenAffine} />;
+      return (
+        account && <OnboardingPage user={account} onOpenAffine={onOpenAffine} />
```
Comment (darkskygit) [inline packages/frontend/core/src/pages/auth.tsx:109]: if not have, redirect to `/`?

File: packages/frontend/core/src/pages/auth.tsx [hunk 2]
```diff
@@ -198,25 +200,3 @@ export const loader: LoaderFunction = async args => {
   }
   return null;
```
Comment (darkskygit) [inline packages/frontend/core/src/pages/auth.tsx:201]: lack of redirect logic for `jumpToExpired`

Comment (forehalo) [reply packages/frontend/core/src/pages/auth.tsx:201]: the logic is only suit for email verify routes, and it's already done

**Review by EYHN — APPROVED** [review]

**Pull request merged**